ISTIO_DOC.md

```diff
@@ -0,0 +1,143 @@
+autoscaler-6c7bf97997-rkd2n            1/1     Running   0          15m
+autoscaler-keda-7f87794cb7-g2wm8       1/1     Running   0          6m32s
+controller-5b54cd98c-bzgpt             1/1     Running   0          15m
+net-istio-controller-c9444c8ff-rwcp5   1/1     Running   0          11m
+net-istio-webhook-66b6b6444c-zpxjg     1/1     Running   0          11m
+webhook-56ffd84996-ck5tm               1/1     Running   0          15m
+```
+
+The testing reported in the [development instructions](./DEVELOPMENT.md) was conducted also for `istio` in simple-mode. 
+If mesh-mode is not required, continue the setup as documented in the main instructions. 
+
+## Mesh-Mode 
+
+Make sure to review the section of the `knative` documentation about installing istio in mesh-mode: [Knative installation doc](https://knative.dev/docs/admin/install/serving/install-serving-with-yaml/#install-a-networking-layer).
+
+Brief instructions are reported below for developing this extension.
+
+Install `istio`:
+
+```bash
+$ kubectl apply -f https://github.com/knative/net-istio/releases/download/knative-v1.16.0/istio.yaml
+$ kubectl apply -f https://github.com/knative/net-istio/releases/download/knative-v1.16.0/net-istio.yaml
+```
+
+### Add knative serving to the mesh
+
+Enable `istio-injection` in the `knative-serving` namespace:
+
+```bash
+$ kubectl label namespace knative-serving istio-injection=enabled
+```
+
+Restart all deployments in the `knative-serving` namespace:
+
+```bash
+$ kubectl rollout restart deploy -n knative-serving
+```
+
+Verify that the pods are now injected with the `istio` sidecar container (note the number of `READY` containers is `2`):
+
+```bash
+kubectl get po -n knative-serving
+NAME                                    READY   STATUS    RESTARTS      AGE
+activator-5754bdb79d-9gpn5              2/2     Running   0             91s
+autoscaler-57d66f69d9-cmv6j             2/2     Running   0             91s
+controller-f48959855-lhrwz              2/2     Running   1 (90s ago)   91s
+net-istio-controller-69858b66f7-qfnq8   1/1     Running   0             91s
+net-istio-webhook-5645569675-jvj2r      2/2     Running   0             91s
+webhook-778946f8-ftjps                  2/2     Running   0             91s
+```
+
+### Configure knative domain 
+
+```bash
+$ kubectl patch configmap/config-domain \
+      --namespace knative-serving \
+      --type merge \
+      --patch '{"data":{"example.com":""}}'
+```
+
+### Install Prometheus and KEDA
+
+In `PERMISSIVE` mode (under the `knative-serving` namespace), clear-text traffic and encrypted traffic are both allowed, with the latter preferred by `istio` when properly configured. The following instructions will not enable Mutual TLS between `knative`, `keda` and `prometheus`. If running in `STRICT` mode, `istio` provides [documentation](https://istio.io/latest/docs/ops/integrations/prometheus/#tls-settings) on how to setup mTLS between "data" plane pods for `prometheus`.
```

Comment:
I think it is useful to have the strict mode as well. I have an example [here](https://github.com/skonto/serverless-guides/blob/main/kserve/minikube/README.md). I will create a PR with some more info so we can finalize content.
Btw did you test encrypted traffic with permissive mode and autoscaling?